java-frontend/src/main/java/org/sonar/java/se/checks/ParameterNullnessCheck.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * SonarQube Java
+ * Copyright (C) 2012-2018 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.java.se.checks;
+
+import com.google.common.collect.Lists;
+import java.util.List;
+import org.sonar.check.Rule;
+import org.sonar.java.resolve.JavaSymbol;
+import org.sonar.java.resolve.Scope;
+import org.sonar.java.se.CheckerContext;
+import org.sonar.java.se.ProgramState;
+import org.sonar.java.se.constraint.ObjectConstraint;
+import org.sonar.java.se.symbolicvalues.SymbolicValue;
+import org.sonar.plugins.java.api.semantic.Symbol;
+import org.sonar.plugins.java.api.tree.Arguments;
+import org.sonar.plugins.java.api.tree.MethodInvocationTree;
+import org.sonar.plugins.java.api.tree.NewClassTree;
+import org.sonar.plugins.java.api.tree.Tree;
+
+import static org.sonar.java.se.ExplodedGraphWalker.EQUALS_METHODS;
+import static org.sonar.java.se.NullableAnnotationUtils.isAnnotatedNullable;
+import static org.sonar.java.se.NullableAnnotationUtils.nonNullAnnotationOnParameters;
+
+@Rule(key = "S4449")
+public class ParameterNullnessCheck extends SECheck {
+
+  @Override
+  public ProgramState checkPreStatement(CheckerContext context, Tree syntaxNode) {
+    ProgramState state = context.getState();
+    if (syntaxNode.is(Tree.Kind.METHOD_INVOCATION)) {
+      MethodInvocationTree mit = (MethodInvocationTree) syntaxNode;
+      checkParameters(mit, mit.symbol(), mit.arguments(), state);
+    } else if (syntaxNode.is(Tree.Kind.NEW_CLASS)) {
+      NewClassTree nct = (NewClassTree) syntaxNode;
+      checkParameters(nct, nct.constructorSymbol(), nct.arguments(), state);
+    }
+    return state;
+  }
+
+  private void checkParameters(Tree syntaxnode, Symbol symbol, Arguments arguments, ProgramState state) {
+    if (!symbol.isMethodSymbol() || arguments.isEmpty()) {
+      return;
+    }
+    JavaSymbol.MethodJavaSymbol methodSymbol = (JavaSymbol.MethodJavaSymbol) symbol;
+    Scope parameters = methodSymbol.getParameters();
+    if (parameters == null) {
+      // FIXME: scope should never be null for parameters (null in default constructors for instance)
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Take the required action to fix the issue indicated by this comment. ([squid:S1134](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1134))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.java%3Ajava&branch=SONARJAVA-2634&issues=AWH1wR9ZW8YHzkWfwHU9&open=AWH1wR9ZW8YHzkWfwHU9)

[ISSUE_KEY]: AWH1wR9ZW8YHzkWfwHU9